state/volume.go

```diff
@@ -1250,16 +1250,6 @@ func (sb *storageBackend) SetVolumeAttachmentInfo(hostTag names.Tag, volumeTag n
 	if _, err := v.Info(); err != nil {
 		return errors.Trace(err)
 	}
-	// Also ensure the machine is provisioned.
-	if _, ok := hostTag.(names.MachineTag); ok {
```

Comment:
Is there code in the domain to replace this functionality? Or a Jira card if not.

Same for blocks below